build-tools/webpack/assets-writer-plugin.js

```diff
@@ -43,50 +43,41 @@ Object.assign( AssetsWriter.prototype, {
 				reasons: false,
 			} );
 
+			function fixupPath( f ) {
+				return path.join( stats.publicPath, f );
+			}
+
+			// Exclude hot update files (info.hotModuleReplacement) and source maps (info.development)
+			function isDevelopmentAsset( name ) {
+				const asset = stats.assets.find( ( a ) => a.name === name );
+				if ( ! asset ) {
+					return false;
+				}
+
+				return asset.info.hotModuleReplacement || asset.info.development;
```

Comment:
Wasn't able to see a single asset where `development` is `true`. Is this intended to be for source maps? If we're not sure, should we keep the pre-existing `*.map` check?

Reply:
Yes, `development: true` is a flag that source maps have. It's reliable, `html-webpack-plugin` also uses it, for the same job as we do here: figure out which compilation assets and how to embed into the app's HTML document.

Here is a webpack source reference where the source maps plugin adds the flag: https://github.com/webpack/webpack/blob/3919c844eca394d73ca930e4fc5506fb86e2b094/lib/SourceMapDevToolPlugin.js#L547-L550